### DISCUSSION TITLE
Movable dynamic inner block for WooCommerce cart/checkout block

### DISCUSSION THREAD

**Discussion by tarunparswani26** [woocommerce/woocommerce#44960] [WooCommerce, Blocks, and Site Editing]
Hi,

I'm encountering difficulties while adding an inner block to the cart/checkout block.

- Trying to create a dynamic inner block i.e. the HTML to be rendered is generated from PHP
- Allowing the inner block to be movable between blocks, enabling users to choose its position.

Currently, I've observed that the inner block is only rendered on the front end when the lock attribute is included in the block.json file, as shown below:
```
"lock": {
			"type": "object",
			"default": {
				"remove": true,
				"move": true
			}
		}
```

I've examined the WooCommerce shipping workshop block code available at [this](https://github.com/woocommerce/wceu23-shipping-workshop-final/) repository. I noticed that similar to my scenario, removing the aforementioned `lock` attribute from the `block.json` file prevents the block from rendering on the front end.

While my ultimate goal is to have the block movable, I am currently considering the option of keeping it locked to a specific parent within the cart/checkout block.

If you have any alternative solutions or suggestions regarding this matter, I would greatly appreciate your input.

**Comment by nielslange** [comment]
Hello @tarunparswani26, 

Thanks for reaching out. You mentioned that you're trying to create a dynamic block that can be moved to different positions, and that the block is visible on the frontend when using
```json
"lock": {
  "type": "object",
  "default": {
    "remove": true,
    "move": true
  }
}
```

Is it correct to say that your primary issue is how to make the inner block movable? If so, the `Terms and Conditions` inner block has no `lock` attribute and can be moved within the Checkout block:

<img width="1008" alt="Screenshot 2023-07-20 at 14 20 15" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/5d23b360-5743-4f2b-9b3a-3d30c57ef03f">

On https://github.com/woocommerce/woocommerce-blocks/blob/trunk/assets/js/blocks/checkout/inner-blocks/checkout-terms-block/block.json, you can find the `block.json` of the `Terms and Conditions` inner block.

If that's still not working as expected, could you share more information about the purpose of your inner block, the way it functions and potentially share your code so that we can look into it?

**Reply by tarunparswani26** [reply]
Hi @nielslange,

Thank you for your suggestion.

I've tried similar things that didn't work but have not explicitly tried the `Terms and Conditions` inner block. Will check it again and get back to you.

**Reply by tarunparswani26** [reply]
Hi @nielslange ,

Did check the `Terms and Conditions` inner block code again and tried to mimic the same in my `inner block` code.

So, if we remove the `lock` attribute, the `inner block` is movable on the `admin` side, however, saving the inner block does not show any content on the `frontend` as trying to display dynamic content on the frontend. 

If I understand it correctly, currently the `Terms and Conditions` inner block is able to display text on the frontend as the text displayed on the backend and frontend is the same, so the `save` function in `edit.tsx` would be saving the same and hence the text is displayed on `frontend`.

In my case, I'm trying to add an `inner block` for displaying user-specific restricted coupons. So, in this case, the `div` on the admin side would be empty, and on frontend, I'm trying to dynamically display the same using the `html` string in `block.js` which is called from `frontend.js`

You can check the sample code of `newsletter subscription` [here](https://github.com/tarunparswani26/example-wc-block) where the admin and frontend rendering `HTML` is different.

Guess, some issue with dynamic inner blocks. Would be also helpful if you have any sample code for it.

Please check and let me know your views.
